Adding TOS and Try It Now strings for en-US

mkoidin opened the pull request:
In order to support an upcoming update to the Pocket panel, adding two new strings to the properties file.

mkoidin committed:
Adding TOS and Try It Now strings for en-US

flodolo commented on locales/en-US/pocket.properties:
```diff
@@ -26,6 +26,8 @@ tagline = Save articles and videos from Firefox to view in Pocket on any device,
 taglinestory_one = Click the Pocket Button to save any article, video or page from Firefox.
 taglinestory_two = View in Pocket on any device, any time.
 tagssaved = Tags Added
+tos = By continuing, you agree to Pocket's <a href="https://getpocket.com/tos?s=ffi&t=tos&tv=panel_tryit">Terms of Service</a> and <a href="https://getpocket.com/privacy?s=ffi&t=privacypolicy&tv=panel_tryit">Privacy Policy</a>
```
Is there any chance to move the actual URLs outside of the string, and use placeholders here? That way you're free to change URLs any time you want, without touching strings.

mixedpuppy replied:
@flodolo do you have a handy example for them?

flodolo replied:
Not in add-ons, but it could be as simple as

`
tos = By continuing, you agree to Pocket's <a href="#1">Terms of Service</a> and <a href="#2">Privacy Policy</a> 
`

And then str.replace the placeholders with the links. Not sure if add-ons have better way to manage variables when accessing strings.

mixedpuppy replied:
I recommended the use of nsIStringBundle.formatStringFromName to do this.

flodolo commented:
Side note: before merging this PR and exposing strings to localizers, I need information on the deadline, and make sure we don't have other strings to expose.

mixedpuppy commented:
@flodolo my plan is to land at least the changes you've seen in bug 1301442 (which has no string or functional changes, just build changes) and uplift to aurora before the merge on Sept 19 for Fx50.  I hope to do the same with the upcoming changes from Pocket, but I expect they will be smaller and easier to uplift in early beta if we miss that date.  In that case, I would still want to uplift these changes within the first couple weeks of beta.

flodolo commented:
Once you land bug 1301442 (note that the removal patch is m-c only, while the other can land on aurora too), you're not an issue for l10n anymore. 

But we still need to tell people how much time they have to translate these strings (we need to give them some time, usually a couple of weeks), when the add-on with these new strings is going to ship.

mixedpuppy commented:
Once I have the full change (code and strings) I'll set a deadline 2 weeks out.

mkoidin committed:
Remove urls and replace with *

mkoidin commented:
@flodolo @mixedpuppy FYI - we've updated this pull request and replaced the hard-coded URLs with \* that will be replaced in the add-on directly.

I <think> that should cover everything you need, but please advise otherwise.

flodolo commented on locales/en-US/pocket.properties:
```diff
@@ -26,6 +26,8 @@ tagline = Save articles and videos from Firefox to view in Pocket on any device,
 taglinestory_one = Click the Pocket Button to save any article, video or page from Firefox.
 taglinestory_two = View in Pocket on any device, any time.
 tagssaved = Tags Added
+tos = By continuing, you agree to Pocket's *Terms of Service* and *Privacy Policy*
```
I advise against doing this, keep the markup to make people understand they're links, just avoid exposing the URL. Not sure what kind of replacement you plan to do, but it's really brittle, unless you plan to double check all localizations.

mkoidin replied:
Okay, we will take a look at revising.

flodolo commented on locales/en-US/pocket.properties:
```diff
@@ -26,6 +26,8 @@ tagline = Save articles and videos from Firefox to view in Pocket on any device,
 taglinestory_one = Click the Pocket Button to save any article, video or page from Firefox.
 taglinestory_two = View in Pocket on any device, any time.
 tagssaved = Tags Added
+tos = By continuing, you agree to Pocket's *Terms of Service* and *Privacy Policy*
```
One more note: as discussed on IRC, if you're working on more string changes, we should expose them all at the same time.

mkoidin replied:
We have no other planned string changes - it's just these two.

flodolo replied:
Sorry for being a pain, but how are you going to distinguish the first link from the second? What if a localization swaps them for any reason? I strongly suggest to use something Mozilla localizers are used to, like `#1` and `#2`.

@mixedpuppy What were the other strings you were talking about on IRC?

mixedpuppy replied:
@flodolo must have been a misunderstanding.  Whatever pocket submits here is what is getting changed.

flodolo replied:
Thanks. If we can agree on the placeholders, I can send the email later today and merge this PR.

mkoidin committed:
Update replacement variables to not include markup

mkoidin commented:
@flodolo @mixedpuppy Okay, strings updated to add back the markup, but leave the actual URL as a variable. Thank you for the input.

As mentioned above, these are the only planned string changes for the update we are submitting to the Pocket panel.

mixedpuppy commented:
Ok, I think the placeholders should follow what we do:

addonwatch.slow=%1$S might be making %2$S run slowly

This is then used in JS like this:

bundle.getFormattedString("addonwatch.slow", [addon.name, brandShortName]);

mkoidin committed:
Adjust string replacement variables

mkoidin commented:
@flodolo @mixedpuppy We have updated the string to use the recommended placeholders:

%1$S and %2$S

We have pushed the related change to the Pocket panel code as well.

flodolo commented:
I'm going to merge this and send the email after Pontoon runs next sync.
